### PR TITLE
Integrate first V3 record: Epoch

### PR DESCRIPTION
**Pull request by timothycoleman**
Added: V3 Epoch integration

- This connects up the Epoch record in the v3 project via the record factory abstraction point
- V3 Epochs are distinguished from V2 epochs with the version field
- V3 is enabled/disabled manually in ClusterVNodeHostedService

### CODE REVIEW

**Review by shaan1337 — COMMENTED** [review]
@timothycoleman can you please rebase the PR? there are 3 extra commits from the previous PRs. thanks!

File: src/EventStore.Core/TransactionLog/LogRecords/LogV3Record.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		public LogV3Record(ReadOnlyMemory<byte> populatedBytes) {
+			Record = new RecordView<T>(populatedBytes);
+		}
+
+		public void WriteTo(BinaryWriter writer) {
+			writer.Write(Record.Bytes.Span);
+		}
+
+		public int GetSizeWithLengthPrefixAndSuffix() {
+			return 8 + Record.Bytes.Length;
```
Comment (shaan1337) [inline src/EventStore.Core/TransactionLog/LogRecords/LogV3Record.cs:39]:
```suggestion
			return 2 * sizeof(int) + Record.Bytes.Length;
```
Just a small optional suggestion - to be more explicit.

**Review by shaan1337 — COMMENTED** [review]
reviewed & tested - looks good to me :+1:
just needs a rebase and there's a minor suggestion

File: src/EventStore.Core/TransactionLog/LogRecords/LogV3Record.cs (hunk 1 above)
Comment (timothycoleman) [reply src/EventStore.Core/TransactionLog/LogRecords/LogV3Record.cs:39]: :+1:

**Comment by timothycoleman** [comment]
great, thanks. i'll rebase when the pr that this one depends on has gone into master (cause i'll have to do it then anyway)

**Review by shaan1337 — APPROVED** [review]

**Comment by timothycoleman** [comment]
@shaan1337 ive rebased it now :)

**Review by shaan1337 — APPROVED** [review]

**Review by YoEight — APPROVED** [review]

**Pull request merged**